File: output/GPT4o-Mini/xCodeEval_remove_4000/Java/Python/a08e45990fceb02145c141e46c41e3e3.py
import sys
import random
from collections import defaultdict

class B:
    ENTER = "+"
    EXIT = "-"
    
    def __init__(self):
        self.rnd = random.Random()
        self.file_name = ""
        self.id33 = []
        self.input_events = []
        self.were_before = set()
        self.used = []
        self.zeros = set()
    
    def solve(self):
        n = self.read_int()
        m = self.read_int()
        
        self.id33 = [True] * n
        self.input_events = [self.Event(self.read_string(), self.read_int() - 1) for _ in range(m)]
        
        self.used = [False] * n
        for input_event in self.input_events:
            if not input_event.enter and not self.used[input_event.index]:
                self.were_before.add(input_event.index)
            self.used[input_event.index] = True
        
        meeting_size = len(self.were_before)
        for event in self.input_events:
            if event.enter:
                if meeting_size != 0:
                    self.id33[event.index] = False
                else:
                    self.zeros.add(event.index)
                meeting_size += 1
            else:
                meeting_size -= 1
                if meeting_size != 0:
                    self.id33[event.index] = False
                else:
                    self.zeros.add(event.index)
        
        if len(self.zeros) > 1:
            for index in self.zeros:
                self.id33[index] = False
        elif len(self.zeros) == 1:
            zero_enter = next(iter(self.zeros))
            if self.were_before and zero_enter not in self.were_before:
                self.id33[zero_enter] = False
        
        id14 = [i for i in range(n) if self.id33[i]]
        print(len(id14))
        print(" ".join(str(leader + 1) for leader in id14))
    
    class Event:
        def __init__(self, type, index):
            self.enter = (type == B.ENTER)
            self.index = index
    
    def read_line(self):
        return sys.stdin.readline().strip()
    
    def read_string(self):
        return self.read_line()
    
    def read_int(self):
        return int(self.read_string())
    
    def run(self):
        try:
            self.solve()
        except Exception as e:
            print(e, file=sys.stderr)
            sys.exit(-1)

if __name__ == "__main__":
    B().run()

File: output/GPT3.5-Turbo/xCodeEval_remove_4000/Java/C/a08e45990fceb02145c141e46c41e3e3.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#define MAX_N 100000
#define MAX_M 100000

typedef struct {
    bool enter;
    int index;
} Event;

int n, m;
bool id33[MAX_N];
Event inputEvents[MAX_M];
bool used[MAX_N];
bool wasBefore[MAX_N];
bool zeros[MAX_N];
int zerosSize;
int meetingSize;

void solve() {
    scanf("%d %d", &n, &m);

    memset(id33, true, sizeof(id33));
    memset(used, false, sizeof(used));
    memset(wasBefore, false, sizeof(wasBefore));
    memset(zeros, false, sizeof(zeros));
    zerosSize = 0;
    meetingSize = 0;

    for (int i = 0; i < m; ++i) {
        char type[2];
        int index;
        scanf("%s %d", type, &index);
        inputEvents[i].enter = (strcmp(type, "+") == 0);
        inputEvents[i].index = index - 1;
    }

    for (int i = 0; i < m; ++i) {
        Event inputEvent = inputEvents[i];
        if (!inputEvent.enter && !used[inputEvent.index]) {
            wasBefore[inputEvent.index] = true;
        }

        used[inputEvent.index] = true;
    }

    for (int i = 0; i < m; ++i) {
        Event event = inputEvents[i];
        if (event.enter) {
            if (meetingSize != 0) {
                id33[event.index] = false;
            } else {
                zeros[event.index] = true;
                zerosSize++;
            }

            meetingSize++;
        } else {
            meetingSize--;

            if (meetingSize != 0) {
                id33[event.index] = false;
            } else {
                zeros[event.index] = true;
                zerosSize++;
            }
        }
    }

    if (zerosSize > 1) {
        for (int i = 0; i < n; ++i) {
            if (zeros[i]) {
                id33[i] = false;
            }
        }
    } else if (zerosSize == 1) {
        int zeroEnter = -1;
        for (int i = 0; i < n; ++i) {
            if (zeros[i]) {
                zeroEnter = i;
                break;
            }
        }

        if (zeroEnter != -1 && wasBefore[zeroEnter] == false) {
            id33[zeroEnter] = false;
        }
    }

    int id14[MAX_N];
    int id14Size = 0;
    for (int i = 0; i < n; ++i) {
        if (id33[i]) {
            id14[id14Size++] = i;
        }
    }

    printf("%d\n", id14Size);
    for (int i = 0; i < id14Size; ++i) {
        printf("%d ", id14[i] + 1);
    }
    printf("\n");
}

int main() {
    solve();

    return 0;
}
